### PR TITLE
Soycode jpm

### PR DESCRIPTION
**Pull request by soycode**
Successfully builds and runs the testing Firefox addon with jpm. Depends on a small change to the `grunt-jpm` package (see https://github.com/rpl/grunt-jpm/pull/3).

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ module.exports = function (grunt) {
               config: 'bump.options.tagMessage',
               type: 'input',
               message: 'Enter a git tag message:',
-              default: 'v%VERSION%'
+            default: 'v%VERSION%'
```
Comment (willscott) [inline Gruntfile.js]: spacing?

File: tasks/build-test-addon.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ module.exports = function (grunt) {
   var glob = require('glob');
   var pkg = require('../package.json');
   var activeReporters = {};
+  grunt.option('firefox-bin', '/usr/bin/firefox');  // TODO more crossplatformy
```
Comment (willscott) [inline tasks/build-test-addon.js]: this means it won't work on mac right now?  maybe worth an equivalent of https://github.com/uProxy/node-chrome-runner

**Comment by willscott** [comment]
Cool! easier than I was fearing :)

File: Gruntfile.js (hunk 1 above)
Comment (soycode) [inline Gruntfile.js]: Fixed, emacs autospacing was oddly zealous there.

File: tasks/build-test-addon.js (hunk 2 above)
Comment (soycode) [inline tasks/build-test-addon.js]: Agreed - this was a hack to make it run for now as for some reason the default was something weird like /usr/lib64/firefox. https://www.npmjs.com/package/fx-runner may be like what you're suggesting, but another fix may be setting the environment variable grunt-jpm pulls more intelligently. In either case I'll resolve before merging (going to wait on getting the grunt-jpm change in regardless).

**Comment by soycode** [comment]
Just to update here - `grunt-jpm` is now fixed, but `fx-runner` (which is actually depended on by jpm) doesn't correctly start on Ubuntu (https://github.com/erikvold/fx-runner/issues/4). Will work on resolving that and then revisit merging this.

**Comment by soycode** [comment]
On second thought, decided to change `.shippable.yml` to have the needed binary flag for tests to at least work there, which is a better place to have the workaround anyway. So, I'd be open for potentially merging this sooner, feedback of course welcome.

**Comment by willscott** [comment]
how long do you expect it to take for fx-runner to get fixed? if a while, we could depend on a fixed fork of the upstream code until it's fixed for real.

**Comment by soycode** [comment]
Been waiting awhile but looks like the fx-runner fix is blocked for the time being. TBH I think the solution in this branch (just hardcoding the location in .shippable.yml) is cleaner than maintaining our own fork of fx-runner, but either way I think it may be a good idea to get this merged in without waiting on fx-runner. LMK your thoughts - thanks!

**Comment by soycode** [comment]
Going ahead with merging, just to avoid staleness on our side. When `fx-runner` is fixed it'll just be a matter of removing the `-b` flag in .shippable.yml.

**Comment by soycode** [comment]
Looks to be failing the same (new) onDisconnect test as Chrome. This is fixed by - https://github.com/freedomjs/freedom/pull/293

It seems to pass the pre-that-change version of the tests intermittently, but will pass more reliably once that change propagates.

**Pull request merged**